posts/2020-08-16-ocp-log-forwarding.adoc

```diff
@@ -0,0 +1,390 @@
+[root@ocp ~]# oc create configmap secure-forward --from-file=secure-forward.conf -n openshift-logging
+----
++
+If you need to import secrets required for the receiver, also run the following command:
++
+[source]
+----
+[root@ocp ~]# oc create secret generic secure-forward --from-file=<certFile>=cert_file_from_fluentd_receiver
+----
++
+
+. Refresh the *fluentd* Pods to apply the secure-forward secret and secure-forward ConfigMap:
++
+[source]
+----
+oc delete pod --selector logging-infra=fluentd
+----
++
+
+=== Setting up Splunk Dashboard
+
+. Go to Search & Reporting. Search for `index="openshift"` for *Option 1* and `source="openshift"` for *Option 2* to view logs from OpenShift Container Platform.
+
+. Download Splunk dashboards for Open Liberty: link:https://github.com/WASdev/sample.dashboards/tree/master/Liberty/Splunk%208[Sample dashboard for Liberty using Splunk]. Import downloaded sample dashboards using *Source* option. Using this dashboard, you can visualize message, trace, and first failure data capture (FFDC) logging data collected from JSON logging in Open Liberty.
+
+image::/img/blog/splunk-dashboard.png[Splunk-Dashboard,width=70%,align="center"]
+
+== External Elasticsearch and Kibana configuration
+Using `elasticsearch` output, you can forward logs to Elasticsearch cluster deployed on your external machine. You run Elasticsearch and Kibana with docker compose. This way, obtaining docker images, setting up configurations and bringing up the clusters can be done  using one YAML file.
+
+. Create `docker-compose.yaml` file for Elasticsearch and Kibana deployment on your external machine.
+* Sample `docker-compose.yaml`:
++
+```
+version: '3'
+services:
+
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:5.6.16
+    ports:
+      - "9200:9200"
+    volumes:
+      - ./elasticsearch/data:/usr/share/elasticsearch/data
+    networks:
+      - elastic
+
+  kibana:
+    image: docker.elastic.co/kibana/kibana:5.6.16
+    ports:
+      - "5601:5601"
+    networks:
+      - elastic
+```
++
+Deploy Elasticsearch and Kibana.
++
+[source]
+----
+[root@ocp ~]# docker-compose up
+----
++
+
+. Download Kibana dashboards for Open Liberty: link:https://github.com/OpenLiberty/open-liberty-operator/tree/master/deploy/dashboards/logging[Sample Kibana dashboards for Open Liberty]. Import downloaded sample dashboards in Kibana Management > Saved Objects.
+
+image::/img/blog/kibana-dashboard.png[Kibana Dashboard,width=70%,align="center"]
+
+== Conclusion
+Application logging is one of the fundamental types of observability. It helps you monitor and find the problems on your servers easily. Using Log Forwarding API, you can use existing external enterprise log collection solutions for OpenShift Container Platform logs. We have now seen two popular log collection solutions: Splunk and Kibana. Both log analysis solutions allow you to aggregate and analyze log events from Open Liberty servers running on OpenShift Cluster Platform. 
```

Comment:
We have now seen two popular log collection solutions: Splunk and Kibana. Both log analysis solutions allow you to aggregate and analyze log events from Open Liberty servers running on OpenShift Cluster Platform. 
->
We have now seen two popular log collection solutions - one based on Fluentd / Splunk and the other based on Fluentd / Elasticsearch / Kibana. Both log analysis solutions allow you to aggregate and analyze log events from Open Liberty servers running on OpenShift Cluster Platform.